### PR TITLE
Improve basic join implementation

### PR DESCRIPTION
**Pull request by radeusgd**
### Pull Request Description

Implements https://www.pivotaltracker.com/story/show/183913232

### Important Notes

Added counts of succeeded/failed tests within a group and global summary, to easier see how many tests failed.

### Checklist

Please include the following checklist in your PR:

- [x] The documentation has been updated if necessary.
- [x] All code conforms to the
      [Scala](https://github.com/enso-org/enso/blob/develop/docs/style-guide/scala.md),
      [Java](https://github.com/enso-org/enso/blob/develop/docs/style-guide/java.md),
      and
      [Rust](https://github.com/enso-org/enso/blob/develop/docs/style-guide/rust.md)
      style guides.
- All code has been tested:
  - [x] Unit tests have been written where possible.
  - [ ] If GUI codebase was changed: Enso GUI was tested when built using BOTH
        `./run ide build` and `./run ide watch`.

### CODE REVIEW

File: distribution/lib/Standard/Table/0.0.0-dev/src/Data/Join_Condition.enso [hunk 1]
```diff
@@ -21,7 +23,8 @@ type Join_Condition
        Arguments:
         - left: A name or index of a column in the left table.
         - right: A name or index of a column in the right table.
-    Equals_Ignore_Case (left : Text | Integer) (right : Text | Integer)
+        - locale: The locale to use for case insensitive comparisons.
+    Equals_Ignore_Case (left : Text | Integer) (right : Text | Integer) (locale : Locale = Locale.default)
```
Comment (jdunkerley) [inline distribution/lib/Standard/Table/0.0.0-dev/src/Data/Join_Condition.enso]: how about we default right to left.
```suggestion
    Equals_Ignore_Case (left : Text | Integer) (right : Text | Integer = left) (locale : Locale = Locale.default)
```

File: distribution/lib/Standard/Table/0.0.0-dev/src/Internal/Join_Helpers.enso [hunk 2]
```diff
@@ -19,7 +19,7 @@ type Join_Condition_Resolver
        `Nothing` will be returned as the conditions indicating that state.
        Besides, a list of redundant columns from equality joins is aggregated
        which can be used to deduplicate them.
-    resolve : Join_Condition | Vector Join_Condition -> Problem_Behavior -> Join_Condition_Resolution
+    resolve : Join_Condition | Vector (Join_Condition | Text) -> Problem_Behavior -> Join_Condition_Resolution
```
Comment (jdunkerley) [inline distribution/lib/Standard/Table/0.0.0-dev/src/Internal/Join_Helpers.enso]: Shouldn't it be `Join_Condition | Text | Vector (Join_Condition | Text)`

Comment (radeusgd) [reply distribution/lib/Standard/Table/0.0.0-dev/src/Internal/Join_Helpers.enso]: Indeed!

File: distribution/lib/Standard/Table/0.0.0-dev/src/Data/Join_Condition.enso (hunk 1 above)
Comment (radeusgd) [reply distribution/lib/Standard/Table/0.0.0-dev/src/Data/Join_Condition.enso]: Sure

**Review by jdunkerley — APPROVED** [review]

**Pull request merged**